tests/Replication2/ModelChecker/ModelChecker.h

```diff
@@ -592,10 +592,9 @@ struct RandomEnumerator {
       // threadSeed) given here are deterministic.
       auto const iters = iterationsLeftToDistribute / (numThreads - thrIdx);
       iterationsLeftToDistribute -= iters;
+      RandomGenerator::initialize(RandomGenerator::RandomType::MERSENNE);
       threads.emplace_back([&driver, initialObserver, initialState, iters,
                             threadSeed = gen(), result = &results[thrIdx]] {
-        // the random device is thread_local
-        RandomGenerator::initialize(RandomGenerator::RandomType::MERSENNE);
```

Comment:
`RandomGenerator::initialize` reset thread_local ptr to null and global enum `type` to mersenne
It no make sense, because 
1) we have data race on `type` and we can init it before threads creation.
2) thread_local ptr initialized by null on thread creation, this thread created right here